app/models/fe/concerns/choice_field_concern.rb

```diff
@@ -126,13 +126,21 @@ def display_response(app=nil)
     def conditional_match(answer_sheet)
       displayed_response = display_response(answer_sheet)
       (is_true(displayed_response) && is_true(conditional_answer)) ||
-        (is_false(displayed_response) && is_false(conditional_answer)) ||
+        (is_response_false(answer_sheet) && is_false(conditional_answer)) ||
         (displayed_response == conditional_answer)
     end
 
+    def is_response_true(answer_sheet)
+      is_true(display_response(answer_sheet))
+    end
+
+    def is_response_false(answer_sheet)
+      is_false(display_response(answer_sheet))
+    end
+
     protected
     def is_true(val)
-      [1,'1',true,'true','Yes','yes'].include?(val) # note: true = anything but false | nil
+      [1,'1',true,'true','Yes','yes'].include?(val)
```

Comment:
I realize you were just cleaning up a wrong comment, but looking at this line it could be simplified to:

`['1','true','yes'].include?(val.to_s.downcase)`